### PR TITLE
Adding static route for cisco-8000 to handle packets across the backplane.

### PR DESCRIPTION
**Pull request by rraghav-cisco**
These changes are needed to make the multi-asic code to work over cisco-8000 platforms.

Cisco-8000 distributed platform uses internal iBGP routing to determine forwarding path for packets destined to other ASICs(same linecard or different linecards) via the Fabric Card. Since the qos-sai setup removes BGP, this routing information is lost. We need to setup a static route in the source-asic so that the packets can be forwarded to the destination ASIC.

When the qos sai script runs, with just a single asic, we can see the routes are all removed from that asic:
root@sfd-vt2-lc0:/home/cisco# ip netns exec asic0 show ip route -n asic0
Codes: K - kernel route, C - connected, S - static, R - RIP,
       O - OSPF, I - IS-IS, B - BGP, E - EIGRP, N - NHRP,
       T - Table, v - VNC, V - VNC-Direct, A - Babel, D - SHARP,
       F - PBR, f - OpenFabric,
       > - selected route, * - FIB route, q - queued route, r - rejected route

C>*1.0.0.0/16 is directly connected, eth1, 04:44:46
C>*3.3.3.3/32 is directly connected, Loopback4096, 04:44:42
C>*10.0.0.0/31 is directly connected, PortChannel102, 04:43:33
C>*10.0.0.4/31 is directly connected, PortChannel104, 04:43:33
C>*10.0.0.8/31 is directly connected, PortChannel106, 04:43:33
C>*10.0.0.12/31 is directly connected, PortChannel108, 04:43:33
C>*10.0.0.16/31 is directly connected, PortChannel1010, 04:43:33
C>*10.0.0.20/31 is directly connected, PortChannel1012, 04:43:33
C>*10.1.0.1/32 is directly connected, Loopback0, 04:44:42
root@sfd-vt2-lc0:/home/cisco# 

In this PR, we add a static route in the source ASIC to the dst_port_ip via a backplane interface IP address as gateway.  This backplane interface IP is detected from the IP address list in the dst ASIC.

### CODE REVIEW

**Comment by abdosi** [comment]
need more details in PR description. Hard to understand why we need this change

**Comment by rraghav-cisco** [comment]
> need more details in PR description. Hard to understand why we need this change

@abdosi , I have updated the PR description. Pls let me know if you have more questions.

File: tests/qos/qos_sai_base.py [hunk 1]
```diff
@@ -1576,3 +1576,44 @@ def resetWatermark(
...
+        # remove when done.
+        if dutTestParams["basicParams"]["sonic_asic_type"] != "cisco-8000":
+            yield
+            return
+        src_asic = get_src_dst_asic_and_duts['src_asic']
+        dst_asic = get_src_dst_asic_and_duts['dst_asic']
+        if src_asic == dst_asic:
+            yield
+            return
+        dst_ip = dutConfig["testPorts"]["dst_port_ip"]
```
Comment (zhixzhu) [inline tests/qos/qos_sai_base.py:1594]: Some testcases use 3 dst ports, need static routes for dst_port_2_ip and dst_port_3_ip as well.

Could you add static routes to different next hops? Like this:
ip netns exec {src_asic} ip route add {dst_port_ip} via {bp_portchannel_ip}
ip netns exec {src_asic} ip route add {dst_port_2_ip} via {bp_portchannel_2_ip}
ip netns exec {src_asic} ip route add {dst_port_3_ip} via {bp_portchannel_3_ip}

**Comment by abdosi** [comment]
please rebase for 202205.

**Comment by abdosi** [comment]
@rraghav-cisco please target it to 202205 branch at priority.

**Comment by rraghav-cisco** [comment]
@abdosi: New PR raised for 202205: 8365

**Comment by abdosi** [comment]
it is covered by another master PR .

**Pull request closed without merging**